src/nnet3bin/nnet3-xvector-get-egs-seg.cc

```diff
@@ -0,0 +1,167 @@
+// nnet3bin/nnet3-xvector-get-egs-seg.cc
+
+// Copyright 2016-2017  Johns Hopkins University (author:  Hang Lyu)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sstream>
+#include "util/common-utils.h"
+#include "nnet3/nnet-example.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+// Process the utt2label file and store it as a map from utt_id to
+// label
+static void ProcessUttToLabelFile(const std::string &utt2label_rxfilename, 
+    unordered_map<std::string, int32, StringHasher> *utt_to_label) {
```

Comment:
The idea is that in future, specially the label can be even vector like posteriors (e.g. TS learning application)